addon/components/g-map/advanced-marker.js

```diff
@@ -0,0 +1,25 @@
+import TypicalMapComponent from './typical-map-component';
+import { toLatLng } from '../../utils/helpers';
+
+export default class AdvancedMarker extends TypicalMapComponent {
+  get name() {
+    return 'advancedMarkers';
+  }
+
+  get newOptions() {
+    if (!this.args.position) {
+      this.options.position = toLatLng(this.args.lat, this.args.lng);
+    }
+
+    return this.options;
+  }
+
+  update(mapComponent) {
+    super.update(mapComponent);
+    mapComponent.position = this.newOptions.position;
+  }
+
```

Comment:
```suggestion
```

Reply:
If I don't do this, the `it updates the advanced marker’s position` test fails.

Reply:
@sandydoo What do you think? ☝️

Reply:
Ah, I see. But if the test is broken, then so is the component!

It looks like the `setOptions` method is not a thing for the advanced marker. I think we'll have to use the following:

```javascript
update(mapComponent) {
  Object.assign(mapComponent, this.newOptions);

  return mapComponent;
}
```